pkg/operator/sync.go

```diff
@@ -126,6 +144,70 @@ func (optr *Operator) syncTerminationHandler(config *OperatorConfig) error {
 	return nil
 }
 
+func (optr *Operator) syncWebhookConfiguration(config *OperatorConfig) error {
+	if err := optr.syncValidatingWebhook(
+		optr.kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations(),
+		mapiv1.MachineValidatingWebhook(config.TargetNamespace),
+		mapiv1.MachineSetValidatingWebhook(config.TargetNamespace),
+	); err != nil {
+		return err
+	}
+
+	return optr.syncMutatingWebhook(
+		optr.kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations(),
+		mapiv1.MachineMutatingWebhook(config.TargetNamespace),
+		mapiv1.MacheSetMutatingWebhook(config.TargetNamespace),
+	)
+}
+
+func (optr *Operator) syncValidatingWebhook(client admissionclientv1.ValidatingWebhookConfigurationInterface, hooks ...admissionregistrationv1.ValidatingWebhook) error {
+	hook := mapiv1.NewValidatingWebhookConfiguration(hooks...)
+	validHook, err := optr.validatingWebhookLister.Get(hook.Name)
+	if apierrors.IsNotFound(err) {
+		_, err = client.Create(context.Background(), hook, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return err
+	}
+	if !equality.Semantic.DeepDerivative(hook.Webhooks, validHook.Webhooks) {
+		updateHook := validHook.DeepCopy()
+		updateHook.Annotations = hook.Annotations
+		updateHook.Webhooks = hook.Webhooks
+		_, err = client.Update(context.Background(), updateHook, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (optr *Operator) syncMutatingWebhook(client admissionclientv1.MutatingWebhookConfigurationInterface, hooks ...admissionregistrationv1.MutatingWebhook) error {
```

Comment:
we are we passing `...admissionregistrationv1.MutatingWebhook` here? we know ahead of time all we need.

Reply:
Again, the same thing - tests. Testenv accepts full webhook configuration, which was applied already. This is weird, but it also means that this `sync` method might be used in testing, as a mean to create a webhook with needed spec, and then spin a new env with the partial spec to test only specific functionality. See  https://github.com/openshift/machine-api-operator/pull/626#discussion_r449657333

Reply:
>This is weird, 

if there's a reason to do something weird please add a comment so everyone looking at the code can understand the reasoning.

Reply:
Ok